ingestion/functions/parsing/thailand/thailand.py

```diff
@@ -0,0 +1,104 @@
+import json
+import os
+import sys
+from datetime import datetime
+
+# Layer code, like parsing_lib, is added to the path by AWS.
+# To test locally (e.g. via pytest), we have to modify sys.path.
+# pylint: disable=import-error
+try:
+    import parsing_lib
+except ImportError:
+    sys.path.append(
+        os.path.join(
+            os.path.dirname(os.path.dirname(os.path.abspath(__file__))),
+            'common/python'))
+    import parsing_lib
+
+
+def convert_date(raw_date):
+    """
+    Convert raw date field into a value interpretable by the dataserver.
+
+    The date is listed in YYYY-mm-dd HH:MM:SS format, but the date filtering API
+    expects mm/dd/YYYYZ format.
+    """
+    date = datetime.strptime(raw_date, "%Y-%m-%d %H:%M:%S")
+    # Some cases are reported using the Buddhist calendar which is 543 years ahead of the Gregorian
```

Comment:
Fascinating! We have 540 years before this would cause any confusion, so this looks good to me.